bin/test_runner.rb

```diff
@@ -58,21 +71,43 @@ def is_up?
   end
 
   def get_line
-    return "#{@stage} #{@name} #{@percepts[:dup]} #{@percepts[:job_status]} #{@percepts[:destroyer_status]}"
+    return "#{@stage} #{@name} #{@percepts[:dup]} #{@percepts[:job_status]} #{@percepts[:destroyer_status]} #{@percepts[:build_id]} #{@percepts[:build]}"
   end
 
   def done?
     @stage == Done || @stage == Failed || @stage == ErrorState
   end
 
-  def update_percepts
+  def init_percepts
     @percepts[:dup]        = is_up?
     @percepts[:job_status] = job_status
     @percepts[:destroyer_status] = destroyer_status
+    @percepts[:build_id]   = job_id
+    @percepts[:build]      = "same"
+  end
 
+  def update_percepts
+    @percepts[:dup]        = is_up?
+    @percepts[:job_status] = job_status
+    @percepts[:destroyer_status] = destroyer_status
+    new_id = job_id
+    delta = new_id.to_i - @percepts[:build_id].to_i
+    unless delta == 0
```

Comment:
These nested conditionals are hard to read and can be simplified with just a single case block:
```
case delta
when 0
  ...
when 1
  ...
else
  # delta not 0 or 1
  raise "ERROR..."
end
```